### PR TITLE
add unit test for ui and logic

### PR DESCRIPTION
**Pull request by Ahmed7osny1**

### CODE REVIEW

File: src/test/kotlin/presentation/suggest/SuggestItalianFoodForLargeGroupUITest.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    private lateinit var viewer: Viewer
+    private lateinit var ui: SuggestItalianFoodForLargeGroupUI
+
+    @BeforeEach
+    fun setUp() {
+        suggestItalianFoodForLargeGroupUseCase = mockk(relaxed = true)
+        viewer = mockk(relaxed = true)
+        ui = SuggestItalianFoodForLargeGroupUI(suggestItalianFoodForLargeGroupUseCase, viewer)
+    }
+
```
Comment (mena-rizkalla) [inline src/test/kotlin/presentation/suggest/SuggestItalianFoodForLargeGroupUITest.kt:25]: Add Tests for Empty List in TestUI too
and catch the exception in UI first

`try {

val meals = suggestItalianFoodForLargeGroupUseCase
    .suggestItalianMealsForLargeGroup()
meals.forEach { meal ->
    viewer.display(
        "- Meal Name: ${meal.name}\n"   )
} catch (e: NoSuchElementException) {
viewer.display("No Italian meals for large groups found.\n")}`

**Review by mosayed01 — APPROVED** [review]
